docs/conf.py

```diff
@@ -109,7 +109,7 @@
 intersphinx_mapping = {
     "python": ("https://docs.python.org/3", None),
     "numpy": ("https://numpy.org/doc/stable/", None),
-    "scipy": ("https://docs.scipy.org/doc/scipy/reference/", None),
+    "scipy": ("https://docs.scipy.org/doc/scipy/html-scipyorg/", None),
```

Comment:
```suggestion
    "scipy": ("https://docs.scipy.org/doc/scipy/", None),
```

From this comment (https://github.com/scipy/scipy/issues/15545#issuecomment-1032808071) posted after you opened this PR, I think this is the proper change new.  I haven't tested it though.